test/unit/mem/buf/src/BufTest.cpp

```diff
@@ -162,27 +162,35 @@ TEMPLATE_LIST_TEST_CASE("memBufConstTest", "[memBuf]", alpaka::test::TestAccs)
 template<typename TAcc>
 static auto testAsyncBufferImmutable(alpaka::Vec<alpaka::Dim<TAcc>, alpaka::Idx<TAcc>> const& extent) -> void
 {
-    using Dev = alpaka::Dev<TAcc>;
-    using Pltf = alpaka::Pltf<Dev>;
-    using Queue = alpaka::test::DefaultQueue<Dev>;
-
-    using Elem = float;
-    using Dim = alpaka::Dim<TAcc>;
-    using Idx = alpaka::Idx<TAcc>;
-
-    Dev const dev = alpaka::getDevByIdx<Pltf>(0u);
-    Queue queue(dev);
-
-    // memory is allocated when the queue reaches this point
-    auto const buf = alpaka::allocAsyncBuf<Elem, Idx>(queue, extent);
-
-    // synchronous operations must wait for the memory to be available
-    alpaka::wait(queue);
-    auto const offset = alpaka::Vec<Dim, Idx>::zeros();
-    alpaka::test::testViewImmutable<Elem>(buf, dev, extent, offset);
+    {
+        using Dev = alpaka::Dev<TAcc>;
+        using Pltf = alpaka::Pltf<Dev>;
+        using Queue = alpaka::test::DefaultQueue<Dev>;
+
+        using Elem = float;
+        using Dim = alpaka::Dim<TAcc>;
+        using Idx = alpaka::Idx<TAcc>;
+
+        Dev const dev = alpaka::getDevByIdx<Pltf>(0u);
+        Queue queue(dev);
+
+        // memory is allocated when the queue reaches this point
+        auto const buf = alpaka::allocAsyncBuf<Elem, Idx>(queue, extent);
+
+        // synchronous operations must wait for the memory to be available
+        alpaka::wait(queue);
+        auto const offset = alpaka::Vec<Dim, Idx>::zeros();
+        alpaka::test::testViewImmutable<Elem>(buf, dev, extent, offset);
+
+        // the buffer will queue the deallocation of the memory when it goes out of scope,
+        // and extend the lifetime of the queue until all memory operations have completed.
+        // delay the end end of the queue to push the buffer deletetion task
+        // after all local refs to queue have been dropped
+        alpaka::enqueue(queue, []() { std::this_thread::sleep_for(std::chrono::microseconds(1000)); });
+        [](auto) {}(std::move(queue));
+    }
 
-    // the buffer will queue the deallocation of the memory when it goes out of scope,
-    // and extend the lifetime of the queue until all memory operations have completed.
+    std::this_thread::sleep_for(std::chrono::microseconds(1200));
```

Comment:
Why is this needed ?

Reply:
This makes sure, the test case does not end before the buffer's deleter ran, so any exceptions thrown during it will show up there. At this point the queue is detached (both on `develop` and in this PR), so it does not synchronize with the end of this scope.